CMakeLists.txt

```diff
@@ -156,12 +156,31 @@ else()
     message(FATAL_ERROR "Could not find msgpack-cxx")
 endif()
 
-# Add abseil-cpp
-set(ABSL_PROPAGATE_CXX_STD ON)
-add_subdirectory(submodules/abseil-cpp)
+# Include dependency settings.
+include("build/deps/cmake-settings/settings.cmake")
+
+find_package(outcome REQUIRED)
+if(outcome_FOUND)
+    message(STATUS "Found outcome.")
+else()
+    message(FATAL_ERROR "Could not find libraries for outcome.")
+endif()
+
+find_package(Catch2 3.8.0 REQUIRED)
+if(Catch2_FOUND)
+    message(STATUS "Found Catch2 ${Catch2_VERSION}.")
+else()
+    message(FATAL_ERROR "Could not find libraries for Catch2.")
+endif()
```

Comment:
@coderabbitai Please create an issue to only build tests and require testing dependencies when `BUILD_TESTING` is set `ON`.